src/state/psarstake/hederaHooks.ts

```diff
@@ -115,12 +118,18 @@ export function useDerivativeHederaSarStake(positionId?: BigNumber) {
       // double check
       if (!isAssociated) return;
 
+      // we need to send 0.1$ in hbar amount to mint
+      const HBARAmount = HBARPrice !== 0 ? 0.1 / HBARPrice : 0;
+
+      const rent = hederaFn.TinyBarToHbar(hederaFn.HBarToTinyBars(HBARAmount.toString()));
```

Comment:
1. lets keep rent in string to keep everything consistent. then when you use it in setPayableAmount you can use it like this `setPayableAmount(Hbar.fromString(rent))`
2. why do you need to do HBarToTinyBars and then TinyBarToHbar ? cant you just use do 
`const rent = HBARAmount.toString()` then as i mentioned in 1st point, you can pass rent as string

Reply:
@prodesert22 looks like 1st point is not fixed